### ISSUE TITLE
Chapter 3: Precision_recall_curve

### ISSUE THREAD

**Issue by CharlesFr** [ageron/handson-ml#87]
The  `cross_val_predict()` returns a shape (60000, 2) which causes `precision_recall_curve()` to fail. It must be converted into (60000, 1) before being used to plot the curve. 

```
y_scores = cross_val_predict(sgd_clf, X_train, y_train_5, cv=3, method="decision_function")
# y_scores.shape -> (60000, 2)

precisions, recalls, thresholds = precision_recall_curve(y_train_5, y_scores[:,1])
# must be converted to (60000, 1) using y_scores[:,1]
```

**Comment by ageron** [comment]
Hi @CharlesFr , thanks for your feedback.

The `cross_val_predict()` issue is actually due to a [bug in Scikit-Learn 0.19](https://github.com/scikit-learn/scikit-learn/issues/9589) which adds an extra first dimension to the result, for no good reason, in some cases (in particular with binary classification and `method="decision_function"`).  I added a workaround in the notebook, where basically I drop the first dimension if I see that the result has two dimensions.

Scikit-Learn 0.19 also caused a couple more issues that I'm going through right now:
* `Pipeline` now refuses a tuple of tuples, it wants a list of tuples. Sigh. I updated the notebooks accordingly.
* `LabelBinarizer` no longer works in pipelines. Granted, it was never designed for handling input features, only labels.  The correct solution will be to use `OneHotEncoder` once it supports strings as categorical input features (they're working on it). In the meantime, I have just created a small subclass that changes the signature of the `fit_transform()` method, and everything works fine (ugly, but it works).

I'm going through all notebooks right now with the latest versions of every library, so far these are the only issues I have found, but I still have a few notebooks to run.

Hope this helps!
